SoundGood/Scenes/Genre/GenreViewController.swift

```diff
@@ -43,18 +51,55 @@ class GenreViewController: UIViewController {
         genres.sort {
             $0.title < $1.title
         }
+        filteredGenres = genres
+    }
+
+    private func navigateToTrackByGenre(with data: Genre) {
+        guard let controller = Storyboards.trackByGenre.instantiateInitialViewController() as? TrackByGenreViewController else { return }
+        controller.genreTitle = data.title
+        guard let view = navigationController?.view else { return }
+        UIView.transition(with: view, duration: 0.2, options: .transitionCrossDissolve, animations: nil, completion: nil)
+        navigationController?.pushViewController(controller, animated: false)
     }
 }
 
 // MARK: - TableView data source
 extension GenreViewController: UITableViewDataSource {
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return genres.count
+        return filteredGenres.count
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell: GenreTableViewCell = genreTableView.dequeueReusableCell(for: indexPath)
-        cell.setup(title: genres[indexPath.row].title)
+        cell.setup(title: filteredGenres[indexPath.row].title)
         return cell
     }
 }
+
+// MARK: - TableView delegate
+extension GenreViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        navigateToTrackByGenre(with: filteredGenres[indexPath.row])
+    }
+}
+
+// MARK: - SearchBar delegate
+extension GenreViewController: UISearchBarDelegate {
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        filteredGenres = filterData(genres, with: searchText)
+        genreTableView.reloadData()
+    }
+
+    private func filterData(_ genres: [Genre], with keyword: String) -> [Genre] {
+        var filtered = [Genre]()
```

Comment:
đoạn code này có thể refactor được đó
if keyword is empty return genres
return genres.filter {
}